pkg/application/ecosystemConfigUseCase.go

```diff
@@ -78,102 +84,132 @@ func (useCase *EcosystemConfigUseCase) ApplyConfig(ctx context.Context, blueprin
 func (useCase *EcosystemConfigUseCase) applyGlobalConfigDiffs(ctx context.Context, globalConfigDiffsByAction map[domain.ConfigAction][]domain.GlobalConfigEntryDiff) error {
 	var errs []error
 
+	globalConfig, err := useCase.globalConfigRepository.Get(ctx)
+	if err != nil {
+		return err
+	}
+
+	updatedEntries := globalConfig.Config
 	entryDiffsToSet := globalConfigDiffsByAction[domain.ConfigActionSet]
-	var entriesToSet = make([]*ecosystem.GlobalConfigEntry, 0, len(entryDiffsToSet))
 	for _, diff := range entryDiffsToSet {
-		entry := &ecosystem.GlobalConfigEntry{
-			Key:   diff.Key,
-			Value: common.GlobalConfigValue(diff.Expected.Value),
-		}
-		entriesToSet = append(entriesToSet, entry)
+		var err error
+		updatedEntries, err = updatedEntries.Set(diff.Key, common.GlobalConfigValue(diff.Expected.Value))
+		errs = append(errs, err)
 	}
 
 	entryDiffsToRemove := globalConfigDiffsByAction[domain.ConfigActionRemove]
-	var keysToDelete = make([]common.GlobalConfigKey, 0, len(entryDiffsToRemove))
 	for _, diff := range entryDiffsToRemove {
-		keysToDelete = append(keysToDelete, diff.Key)
+		updatedEntries = updatedEntries.Delete(diff.Key)
 	}
 
-	errs = append(errs, callIfNotEmpty(ctx, entriesToSet, useCase.globalConfigRepository.SaveAll))
-	errs = append(errs, callIfNotEmpty(ctx, keysToDelete, useCase.globalConfigRepository.DeleteAllByKeys))
+	if len(entryDiffsToSet) != 0 || len(entryDiffsToRemove) != 0 {
+		_, err = useCase.globalConfigRepository.Update(ctx, config.GlobalConfig{Config: updatedEntries})
+		errs = append(errs, err)
+	}
 
 	return errors.Join(errs...)
 }
 
-func (useCase *EcosystemConfigUseCase) applyDoguConfigDiffs(ctx context.Context, doguConfigDiffsByAction map[domain.ConfigAction]domain.DoguConfigDiffs) error {
+func (useCase *EcosystemConfigUseCase) applyDoguConfigDiffs(
```

Comment:
🚫 **[checkstyle]** <dupl> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>113-152 lines are duplicate of `pkg/application/ecosystemConfigUseCase.go:154-193`
<!-- __reviewdog__:ChBhMTc0MTRlNmYwZTljZDczEgpjaGVja3N0eWxl -->